src/taco-dependency-installer/test/javaJdkInstaller.ts

```diff
@@ -0,0 +1,344 @@
+                    "error.name": { isPii: false, value: "Error" },
+                    "install.time": { isPii: false, value: "2000" },
+                    lastStepExecuted: { isPii: false, value: "install" },
+                    step: { isPii: false, value: "install" },
+                    time: { isPii: false, value: "3000" }
+                }
+            ];
+
+            return telemetryGeneratedShouldBe(expectedTelemetry, /didn\'t exit cleanly/, done);
+        });
+
+        it("generates telemetry error when the detach dmg command fails", (done: MochaDone) => {
+            // child_process.exec call will only fail for the install packages call
+            childProcessModule.exec = fakeExecToTestPackageInstallerFailure;
+            childProcessExecBehavior = ChildProcessExecBehavior.DetachingDmgFileFailsScenario;
+
+            var expectedTelemetry: TacoUtility.ICommandTelemetryProperties[] = [
+                {
+                    "initialStep.time": { isPii: false, value: "2000" },
+                    step: { isPii: false, value: "initialStep" }
+                },
+                {
+                    "error.description": { isPii: false, value: "ErrorOnChildProcess on detachDmg" },
+                    "install.time": { isPii: false, value: "2000" },
+                    lastStepExecuted: { isPii: false, value: "install" },
+                    step: { isPii: false, value: "install" },
+                    time: { isPii: false, value: "3000" }
+                }
+            ];
+
+            return telemetryGeneratedShouldBe(expectedTelemetry, /Detach dmg command failed to execute/, done);
+        });
+    });
+
+    describe("installation in windows", () => {
+        beforeEach(() => {
+            fakeProcess.fakeWindows();
+            steps.install = true; // We only test this step on this test
+        });
+
+        it("generates telemetry error if there is no install location", (done: MochaDone) => {
+            installTo = ""; // We don't have an install location
+
+            var expectedTelemetry: TacoUtility.ICommandTelemetryProperties[] = [
+                {
+                    "initialStep.time": { isPii: false, value: "2000" },
+                    step: { isPii: false, value: "initialStep" }
+                },
+                {
+                    "error.description": {
+                        isPii: false,
+                        value: "InstallDestination needed on installWin32"
+                    },
+                    "install.time": { isPii: false, value: "2000" },
+                    lastStepExecuted: { isPii: false, value: "install" },
+                    step: { isPii: false, value: "install" },
+                    time: { isPii: false, value: "3000" }
+                }
+            ];
+
+            return telemetryGeneratedShouldBe(expectedTelemetry, /An installation destination is needed for this dependency/, done);
```

Comment:
`/An installation destination is needed for this dependency/` Is it possible to use test mode for the ResourceManager here? That way we only work with string IDs